test/shim-monitor.spec.ts

```diff
@@ -63,6 +66,49 @@ describe('ShimMonitor unit tests', () => {
     globalMock.digitalData.cart.cartID = 'cart-5678';
   });
 
+  it('it should emit args from function calls', (done) => {
```

Comment:
It's probably also worth testing that functions which throw exceptions are nicely handled.

Reply:
I went back and looked at the `it should emit args but trap function exceptions` test case:

- The `done` in the event listener tests whether the actual handler was invoked, which should always be the case since `emit` is called prior to the function's `apply`.  `window.dispatchEvent` executes synchronously so `done()` gets called and the next bullet point's code runs.
- The `expect(ret).to.eq(null);` executes next, and this checks sequential lines of code are not impeded by an exception.

The `emit` body in Monitor is wrapped in a try catch block.  I went back and wrote another test `it should emit args and not fail because of dispatch related exceptions` which ensures that if the handler or DLO code were to throw an exception that they are caught and the function call is not impeded.  This is verified by `expect(ret).to.eq(true);` and the fact that the test exits.